utils/check_repo.py

```diff
@@ -0,0 +1,269 @@
+spec = importlib.util.spec_from_file_location(
+    "transformers",
+    os.path.join(PATH_TO_TRANSFORMERS, "__init__.py"),
+    submodule_search_locations=[PATH_TO_TRANSFORMERS],
+)
+transformers = spec.loader.load_module()
+
+
+# If some modeling modules should be ignored for all checks, they should be added in the nested list
+# _ignore_module of this function.
+def get_model_modules():
+    """ Get the model modules inside the transformers library. """
+    _ignore_module = [
+        "modeling_auto",
+        "modeling_encoder_decoder",
+        "modeling_marian",
+        "modeling_mmbt",
+        "modeling_outputs",
+        "modeling_retribert",
+        "modeling_utils",
+        "modeling_transfo_xl_utilities",
+        "modeling_tf_auto",
+        "modeling_tf_outputs",
+        "modeling_tf_pytorch_utils",
+        "modeling_tf_utils",
+        "modeling_tf_transfo_xl_utilities",
+    ]
+    modules = []
+    for attr_name in dir(transformers):
+        if attr_name.startswith("modeling") and attr_name not in _ignore_module:
+            module = getattr(transformers, attr_name)
+            if inspect.ismodule(module):
+                modules.append(module)
+    return modules
+
+
+def get_models(module):
+    """ Get the objects in module that are models."""
+    models = []
+    model_classes = (transformers.PreTrainedModel, transformers.TFPreTrainedModel)
+    for attr_name in dir(module):
+        if "Pretrained" in attr_name or "PreTrained" in attr_name:
+            continue
+        attr = getattr(module, attr_name)
+        if isinstance(attr, type) and issubclass(attr, model_classes) and attr.__module__ == module.__name__:
+            models.append((attr_name, attr))
+    return models
+
+
+# If some test_modeling files should be ignored for the check models are tested, they should be added in the
+# nested list _ignore_module of this function.
+def get_model_test_files():
+    """ Get the model test files."""
+    _ignore_files = [
+        "test_modeling_common",
+        "test_modeling_encoder_decoder",
+        "test_modeling_marian",
+        "test_modeling_mbart",
+        "test_modeling_tf_common",
+    ]
+    test_files = []
+    for filename in os.listdir(PATH_TO_TESTS):
+        if (
+            os.path.isfile(f"{PATH_TO_TESTS}/{filename}")
+            and filename.startswith("test_modeling")
+            and not filename[:-3] in _ignore_files
+        ):
+            test_files.append(filename)
+    return test_files
+
+
+# If some doc files should be ignored for the check models are documented, they should be added in the
+# nested list _ignore_module of this function.
+def get_model_doc_files():
+    """ Get the model doc files."""
+    _ignore_files = [
+        "auto",
+        "dialogpt",
+        "marian",
+        "retribert",
+    ]
+    doc_files = []
+    for filename in os.listdir(PATH_TO_DOC):
+        if os.path.isfile(f"{PATH_TO_DOC}/{filename}") and not filename[:-4] in _ignore_files:
```

Comment:
```suggestion
        if os.path.isfile(f"{PATH_TO_DOC}/{filename}") and not filename[:-4] in _ignore_modules:
```
(nitpick) Also, with `filename[:-4]` do you want to strip `.rst`? `os.path.splitext(filename)[0]` looks more readable to me.